platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/InterruptibleCompletableFuture.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.merkledb;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+
+/**
+ * A wrapper around {@link CompletableFuture} that supports interruption
+ * when the CompletableFuture is canceled.
+ *
+ */
+public class InterruptibleCompletableFuture<T> {
+
+    /** The wrapped CompletableFuture */
+    private volatile CompletableFuture<T> completableFuture;
```

Comment:
Mostly, for the sake of simplicity. I may be missing something but I don't see a straightforward way of utilizing static methods of `java.util.concurrent.CompletableFuture` in case I extend `CompletableFuture`. That is, these static methods will produce instances of `CompletableFuture`, so get instances of `InterruptibleCompletableFuture` I'd have to re-implement these static methods as well, and this approach looks quite complicated.